app/src/main/java/de/westnordost/streetcomplete/view/LaurelWreath.kt

```diff
@@ -0,0 +1,94 @@
+package de.westnordost.streetcomplete.view
+
+import android.content.res.Resources
+import android.graphics.Canvas
+import android.graphics.ColorFilter
+import android.graphics.Paint
+import android.graphics.PixelFormat
+import android.graphics.RectF
+import android.graphics.drawable.Drawable
+import androidx.core.graphics.withRotation
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.ktx.flipHorizontally
+import de.westnordost.streetcomplete.util.ktx.getBitmapDrawable
+
+/*
+Drawable providing decoration, suitable for a circular background
+100 and more: fully grown wreath with all pretty elements
+99 to 10: may be losing elements as it gets smaller
+below: no decorative styling at all
+ */
+class LaurelWreath(val resources: Resources, private val percentageOfGrowth: Int) : Drawable() {
+    val laurelLeafOnStalk = resources.getBitmapDrawable(R.drawable.ic_laurel_leaf_rotated)
+    val laurelStalk = resources.getBitmapDrawable(R.drawable.ic_laurel_leaf_stalk)
+    val horizontalEndingLeaf = resources.getBitmapDrawable(R.drawable.ic_laurel_leaf_ending)
+
+    private val antiAliasPaint: Paint = Paint().apply {
+        isAntiAlias = true
+        isFilterBitmap = true
+    }
+
+    override fun draw(canvas: Canvas) {
+        // Get the drawable's bounds
+        val canvasWidth: Int = bounds.width()
+        val canvasHeight: Int = bounds.height()
+        val circleRadius: Float = Math.min(canvasWidth, canvasHeight).toFloat() / 2f
+
+        val niceSubtleGreen: Paint = Paint().apply { setARGB(255, 152, 184, 126) }
```

Comment:
should be a private field

Reply:
Are you sure that it should be moved out of that function?

Reply:
I am just parroting what the IDE warns about. Android Studio warns about creation of objects in the `draw` function for performance reasons. Since this paint is always the same, it also makes sense.

Reply:
> It's weird that your Android studio didn't display a warning there? Did you switch off warnings?

I switched off some that were consistently annoying and useless. I can enable them again, and triage them again - maybe they improved something.